packages/client/lib/rpc/modules/engine.ts

```diff
@@ -291,7 +310,15 @@ export class Engine {
     const common = this.config.chainCommon
 
     try {
-      await findBlock(toBuffer(parentHash), this.validBlocks, this.chain)
+      const block = await findBlock(toBuffer(parentHash), this.validBlocks, this.chain)
+      const validTerminalBlock = await validateTerminalBlock(block, this.chain)
```

Comment:
I think this check should only be done if this parent block is "pre-merge", else this check will fail for a post-merge block